src/openrct2-ui/windows/Scenery.cpp

```diff
@@ -1405,6 +1405,21 @@ bool WindowScenerySetSelectedItem(const ScenerySelection& scenery)
     return result;
 }
 
+void WindowScenerySetSelectedTab(const ObjectEntryIndex sceneryGroupIndex)
+{
+    const auto* tabInfo = GetSceneryTabInfoForGroup(sceneryGroupIndex);
+    if (tabInfo == nullptr)
+    {
+        tabInfo = &_tabEntries.back();
+    }
+    const auto tabId = std::distance(&*_tabEntries.cbegin(), tabInfo);
+    auto* window = window_find_by_class(WC_SCENERY);
+    if (window != nullptr)
+    {
+        window_event_mouse_down_call(window, WIDX_SCENERY_TAB_1 + tabId);
```

Comment:
This seems a bit out of place to call an event handler manually from outside the message/input handler. Maybe its better to move the functional of the event code to the window and call that directly.

Reply:
Yeah i wasn't sure what the best option was. We use this mechanism in quite a few windows

Reply:
I guess we'll best leave it be as of now since it was done this way prior. Something to keep in mind however